### PR TITLE
Use snakemake to manage mavis scheduling

### PR DESCRIPTION
**Pull request by creisle**
- replace mavis scheduler with snakemake
- use JSON config for snakemake instead of many command line arguments

Note: This will set up for the next PR which will dockerize the whole thing to make future extensions of mavis simpler

### CODE REVIEW

**Comment by creisle** [comment]
Ignore the coverage checks for now. When this is run locally the coverage is 89%. I'm not sure why codecov isn't picking up all the coverage but I am looking into it. I think it has to do with multi-processing during tests

File: docs/tutorials/mini.md [hunk 1]
```diff
@@ -14,112 +14,24 @@ installed
...
-                        MA_mock-A47933_batch-s4W2Go4tinn49nkhSuusrE-2 (dqAuhhcVKejDvHGBXn22xb) is COMPLETED
-                        MA_mock-A47933_batch-s4W2Go4tinn49nkhSuusrE-3 (eB69Ghed2xAdp2VRdaCJBf) is COMPLETED
-[2018-06-01 13:15:28] pairing
-                        MP_batch-s4W2Go4tinn49nkhSuusrE (6LfEgBtBsmGhQpLQp9rXmi) is COMPLETED
-[2018-06-01 13:15:28] summary
-                        MS_batch-s4W2Go4tinn49nkhSuusrE (HDJhXgKjRmseahcQ7mgNoD) is COMPLETED
-                        rewriting: output_dir/build.cfg
-                        run time (hh/mm/ss): 0:00:00
-                        run time (s): 0
+snakemake -j 1 --configfig tests/mini-tutorial.config.json
```
Comment (calchoo) [inline docs/tutorials/mini.md:33]: typo in snakemake command

```
snakemake -j 1 --config tests/mini-tutorial.config.json
```

**Review by calchoo — APPROVED** [review]
Code changes look good, just a small typo that can be fixed later on

File: docs/tutorials/mini.md (hunk 1 above)
Comment (creisle) [reply docs/tutorials/mini.md:33]: good catch, will include in the docs update PR

**Review by dustinbleile — APPROVED** [review]

**Pull request merged**